internal/binder/function/funcs_array.go

```diff
@@ -105,4 +111,327 @@ func registerArrayFunc() {
 			return ValidateLen(2, len(args))
 		},
 	}
+
+	builtins["array_remove"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array, ok := args[0].([]interface{})
+			res := make([]interface{}, 0, len(array))
+			if !ok {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+			for _, item := range array {
+				if item != args[1] {
+					res = append(res, item)
+				}
+			}
+			return res, true
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+
+	builtins["array_last_position"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array, ok := args[0].([]interface{})
+			if !ok {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+
+			for i := len(array) - 1; i >= 0; i-- {
+				if array[i] == args[1] {
+					return i + 1, true
+				}
+			}
+
+			return 0, true
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+
+	builtins["array_contains_any"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array1, ok1 := args[0].([]interface{})
+			if !ok1 {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+			array2, ok2 := args[1].([]interface{})
+			if !ok2 {
+				return errorArraySecondArgumentNotArrayError, false
+			}
+
+			for _, a := range array1 {
+				for _, b := range array2 {
+					if a == b {
+						return true, true
+					}
+				}
+			}
+
+			return false, true
+		},
+		val: func(ctx api.FunctionContext, args []ast.Expr) error {
+			return ValidateLen(2, len(args))
+		},
+	}
+
+	builtins["array_intersect"] = builtinFunc{
+		fType: ast.FuncTypeScalar,
+		exec: func(ctx api.FunctionContext, args []interface{}) (interface{}, bool) {
+			array1, ok1 := args[0].([]interface{})
+			if !ok1 {
+				return errorArrayFirstArgumentNotArrayError, false
+			}
+			array2, ok2 := args[1].([]interface{})
+			if !ok2 {
+				return errorArraySecondArgumentNotArrayError, false
+			}
+			intersection := []interface{}{}
```

Comment:
I think we can let max(len(array1),len(array2)) as the capacity of `intersection` here